lesson13/script.js

```diff
@@ -0,0 +1,86 @@
+function showLoadingImg(){
+    const div = document.createElement('div');
+    const loadingImg = document.createElement('img');
+    div.id = "js-loading";
+    loadingImg.src = "img/loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.body.insertAdjacentElement('afterbegin',div).appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+async function fetchdata(){
+    const endpoint = "https://api.json-generator.com/templates/szdgGQcOLXuk/data?access_token=hu4bc7qh9znx2m8f53mn4mz2hryvdntkavwbw8j0";
+    const response = await fetch(endpoint);
+    if(!response.ok){
+    throw new Error(`${response.status}:${response.text}`);
+    }
+    const json = await response.json();
+    const data = await json.data;
+    return data;
+}
+
+function renderList(result) {
+    const fragment = document.createDocumentFragment();
+    const ul = document.getElementById('js-list');
+        for(const list of result){
+            const li = document.createElement('li');
+            li.className = "list-item";
+            const a = document.createElement('a');
+            const img = document.createElement('img');
+            a.textContent = list.text;
+            a.href = `/${list.a}`;
+            img.src = list.img;
+            img.alt = list.alt;
+            fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
+        }
+    ul.appendChild(fragment);
+}
+
+async function init(){
+    showLoadingImg();
+    try{
+        const result = await fetchdata();
+        if (!result) {
+            return;
+        }
+        renderList(result);
+    }catch(error){
+        document.getElementById('js-list').textContent = error;
+        console.error(error);
+    }finally{
+        removeLoadingImg();
+    }
+}
+
+const modal = document.getElementById('js-modal');
+const openButton = document.getElementById('js-open-btn');
+const closeButton = document.getElementById('js-close-btn');
+const requestButton = document.getElementById('js-request-btn');
+
+function openModalWindow(){   
+    openButton.addEventListener('click',function () {
+        modal.classList.add('is-show');
+        this.classList.add('is-deactive');
+    },false);
+}
+
+function closeModalWindow(){ 
+    closeButton.addEventListener('click', function () {
+        modal.classList.remove("is-show");
+        openButton.classList.remove('is-deactive');
+    },false);
+}
```

Comment:
Q
Why is the click event wrapped in a function?
I was defining it without wrapping it up and if you have an optimal reason I would like to hear it.

Reply:
Because I thought that the purpose of the click event was easy to understand at a glance.

I think it will be easier to understand what the open button opens, what the close button closes, and what the request button requests.

Reply:
IMO
If the purpose is to make it easier to understand, I think it would be better to cut out only the process instead of wrapping each click event.
[sandbox](https://codesandbox.io/s/js-lesson13-forked-6ybk3t?file=/script.js)
```js
function openModalWindow() {
  modal.classList.add("is-show");
  this.classList.add("is-hide");
}
function closeModalWindow() {
  modal.classList.remove("is-show");
  openButton.classList.remove("is-hide");
}
function requestData() {
  modal.classList.remove("is-show");
  init();
}
closeButton.addEventListener("click", closeModalWindow, false);
requestButton.addEventListener("click", requestData, false);
openButton.addEventListener("click", openModalWindow, false);
```

Reply:
Thank you for telling me how to make it easier to understand!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/15/commits/1a85e05cc35798e9a0411b5a74e1dc6661050eea